graphql_client_codegen/src/lib.rs

```diff
@@ -43,83 +44,122 @@ impl Display for GeneralError {
 impl std::error::Error for GeneralError {}
 
 type BoxError = Box<dyn std::error::Error + Send + Sync + 'static>;
-type CacheMap<T> = std::sync::Mutex<BTreeMap<std::path::PathBuf, T>>;
+type CacheMap<K, V> = std::sync::Mutex<BTreeMap<K, V>>;
+type QueryDocument = graphql_parser::query::Document<'static, String>;
 
 lazy_static! {
-    static ref SCHEMA_CACHE: CacheMap<schema::Schema> = CacheMap::default();
-    static ref QUERY_CACHE: CacheMap<(String, graphql_parser::query::Document<'static, String>)> =
+    static ref SCHEMA_CACHE: CacheMap<std::path::PathBuf, Schema> = CacheMap::default();
+    static ref QUERY_CACHE: CacheMap<std::path::PathBuf, (String, QueryDocument)> =
```

Comment:
Maybe `CacheMap<String, QueryDocument>` if we want to separate caching from reading from query file?

Reply:
What was your thinking behind this? The only scenario I can think of is that a developer has two files with identical content and it would allow the crate to skip GQL parsing. I am not sure that scenario is common enough to pay for the complexity. WDYT?

Reply:
Ah okay, I wasn't sure if it was desirable to cache the `QueryDocument` if the string didn't change. I'll remove this if it's only important to track file changes.

Reply:
The query cache is there because you can have multiple operations (so multiple structs, with multiple derives) for a single GraphQL executable document (= a single file). There are tests for this.